include/counter.h

```diff
@@ -33,12 +33,11 @@ struct counter_alarm_cfg;
 
 /** @brief Alarm callback
  *
- * @param dev     Pointer to the device structure for the driver instance.
- * @param cfg     Original structure owning this handler.
- * @param ticks   Counter value that triggered the callback.
+ * @param user_data User data.
+ * @param chan_id   Channel ID.
+ * @param ticks     Counter value that triggered the callback.
  */
-typedef void (*counter_alarm_callback_t)(struct device *dev,
-					 const struct counter_alarm_cfg *cfg,
+typedef void (*counter_alarm_callback_t)(void *user_data, u8_t chan_id,
 					 u32_t ticks);
 
 /** @brief Alarm callback structure.
```

Comment:
The sentence "Beware such structure should not be allocated on stack." on line 46 should be removed. We could mention in the description of `counter_set_channel_alarm()` function that the `alarm_cfg` parameter does not require static memory allocation.